scripts/ray/dev_tpu.py

```diff
@@ -64,10 +66,129 @@
 # script instead of launching a driver job. This confuses cloudpickle for some reason.
 from src.marin.cluster import ray as ray_utils
 from src.marin.cluster.config import RayClusterConfig, find_config_by_region
+from src.marin.run.vars import LIBTPU_INIT_ARGS
 from src.marin.utils import _hacky_remove_tpu_lockfile
 
 logger = logging.getLogger(__name__)
 
+# Default environment variables to forward to remote
+DEFAULT_ENV_VARS = [
+    "HF_TOKEN",
+    "WANDB_API_KEY",
+    "MARIN_PREFIX",
+    "OPENAI_API_KEY",
+    "GCLOUD_PROJECT",
+    "GCLOUD_TOKEN_PATH",
+    "RAY_ADDRESS",
+    "RAY_API_SERVER_ADDRESS",
+    "RAY_DASHBOARD_ADDRESS",
+    "WANDB_MODE",
+    "RUN_ID",
+]
+
+
+def build_env_dict(env_vars: list[str], extra_env: list[str] = None, forward_all: bool = False) -> dict[str, str]:
+    """Build environment variable dictionary for forwarding."""
+    # Start with all environment variables if requested, otherwise just defaults
+    if forward_all:
+        env_dict = dict(os.environ)
+    else:
+        env_dict = {}
+        for var in env_vars:
+            if value := os.environ.get(var):
+                env_dict[var] = value
+
+    if "LIBTPU_INIT_ARGS" not in env_dict:
+        env_dict["LIBTPU_INIT_ARGS"] = LIBTPU_INIT_ARGS
+
+    CONFIG_FILES = [".levanter.yaml", ".marin.yaml", ".config"]
+    for config_file in CONFIG_FILES:
+        if os.path.exists(config_file):
+            logger.info(f"Injecting environment variables from {config_file}")
+            try:
+                config_yaml = yaml.safe_loads(open(config_file).read())
+            except Exception as e:
+                logger.warning(f"Failed to load config from environment {e}")
```

Comment:
I believe L113 below will `raise NameError` on `config_yaml` if this branch is taken.